// @g-loot/react-tournament-brackets/dist/cjs/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createTheme = exports.SVGViewer = exports.MATCH_STATES = exports.Match = exports.DoubleEliminationBracket = exports.SingleEliminationBracket = void 0;
const single_elim_bracket_1 = __importDefault(require("./bracket-single/single-elim-bracket"));
exports.SingleEliminationBracket = single_elim_bracket_1.default;
const double_elim_bracket_1 = __importDefault(require("./bracket-double/double-elim-bracket"));
exports.DoubleEliminationBracket = double_elim_bracket_1.default;
const match_1 = __importDefault(require("./components/match"));
exports.Match = match_1.default;
const match_states_1 = require("./core/match-states");
Object.defineProperty(exports, "MATCH_STATES", { enumerable: true, get: function () { return match_states_1.MATCH_STATES; } });
const svg_viewer_1 = __importDefault(require("./svg-viewer"));
exports.SVGViewer = svg_viewer_1.default;
const themes_1 = require("./themes/themes");
Object.defineProperty(exports, "createTheme", { enumerable: true, get: function () { return themes_1.createTheme; } });
